app.js

```diff
@@ -314,23 +319,33 @@ const completeRecoveryInit = async ctx => {
     const securityCode = password.randomPassword({ length: SECURITY_CODE_DIGITS, characters: password.digits });
 
     const { detail, kind } = method;
-    const [recoveryMethod] = await RecoveryMethodService.listRecoveryMethods({
+    const [recoveryMethod] = await recoveryMethodService.listRecoveryMethods({
```

Comment:
This was one I missed in the refactor because I was trying for as close to a 1:1 change as possible. Since all parameters are passed in it should indeed be able to call a `getRecoveryMethod`. It's also a bit of defensive programming since I wasn't able to verify whether the full set of parameters is always included in the request here, but probably unnecessary.

Reply:
Turns out we always generate `seedPhrase` in the client before calling the endpoints that invoke this function, so `publicKey` will exist when querying for recovery methods. I'll replace this with an upsert to simplify the logic in this method 👍